### PR TITLE
NAS-123228 / 23.10 / Send `zfs.snapshot.query` events

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/zfs_/snapshot.py [hunk 1]
```diff
@@ -300,4 +302,6 @@ def do_delete(self, id, options):
 
             raise CallError(str(e))
         else:
+            # TODO: Events won't be sent for child snapshots in recursive delete
+            self.middleware.send_event(f'{self._config.namespace}.query', 'REMOVED', id=id)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/zfs_/snapshot.py:306]: @yocalebo should I create an issue for UI team so they handle that on their side? (they already have a list of snapshots when they receive this event)

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/zfs_/snapshot.py:306]: @themylogin yes, please

**Review by yocalebo — APPROVED** [review]

**Comment by yocalebo** [comment]
retest this please

File: src/middlewared/middlewared/plugins/zfs_/snapshot.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/zfs_/snapshot.py]: https://ixsystems.atlassian.net/browse/NAS-123278

**Comment by themylogin** [comment]
time 2:00

**Pull request merged**